### PR TITLE
mig deformable_conv to deform_conv2d

### PR DESCRIPTION
**Pull request by baiyfbupt**
### PR types

Function optimization
### PR changes

APIs
### Describe

- 删除`paddle.fluid.layers.deformable_conv -> paddle.static.nn.deformable_conv`的alias，重新实现`paddle.static.nn.deform_conv2d`接口，修改英文文档和示例代码，改成使用2.0API。
- 修改输入参数名：
  - `paddle.fluid.layers.deformable_conv`函数定义

    ```python
    paddle.fluid.layers.deformable_conv(input, offset, mask, num_filters, filter_size, stride=1, padding=0, dilation=1, groups=None, deformable_groups=None, im2col_step=None, param_attr=None, bias_attr=None, modulated=True, name=None)
    ```

  - `paddle.static.nn.deform_conv2d`函数定义
    ```python
    paddle.static.nn.deform_conv2d(x, offset, mask, num_filters, filter_size, stride=1, padding=0, dilation=1, groups=1, deformable_groups=1, im2col_step=1, weight_attr=None, bias_attr=None, name=None)
    ```

预览效果见https://github.com/PaddlePaddle/Paddle/pull/27841#issuecomment-706913882

### CODE REVIEW

**Comment by baiyfbupt** [comment]
image
image
image

**Review by jzhang533 — APPROVED** [review]
lgtm

File: python/paddle/static/nn/common.py [hunk 1]
```diff
@@ -163,3 +163,180 @@ def fc(x,
...
+
+            H_{out}&= \\frac{(H_{in} + 2 * paddings[0] - (dilations[0] * (H_f - 1) + 1))}{strides[0]} + 1 \\\\
+            W_{out}&= \\frac{(W_{in} + 2 * paddings[1] - (dilations[1] * (W_f - 1) + 1))}{strides[1]} + 1
+
+    Args:
+        x (Tensor): The input image with [N, C, H, W] format. A Tensor with type
+            float32, float64.
+        offset (Tensor): The input coordinate offset of deformable convolution layer.
+            A Tensor with type float32, float64.
+        Mask (Tensor, Optional): The input mask of deformable convolution layer.
```
Comment (TCChenlong) [inline python/paddle/static/nn/common.py:232]: Mask -> mask

File: python/paddle/static/nn/common.py [hunk 2]
```diff
@@ -163,3 +163,180 @@ def fc(x,
...
+            A Tensor with type float32, float64. It should be None when you use
+            deformable convolution v1.
+        num_filters(int): The number of filter. It is as same as the output
+            image channel.
+        filter_size (int|tuple): The filter size. If filter_size is a tuple,
+            it must contain two integers, (filter_size_H, filter_size_W).
+            Otherwise, the filter will be a square.
+        stride (int|tuple): The stride size. If stride is a tuple, it must
```
Comment (TCChenlong) [inline python/paddle/static/nn/common.py:240]: stride -> im2col_step 都需要加上"optional"

File: python/paddle/static/nn/common.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/static/nn/common.py:232]: will fix in next PR

File: python/paddle/static/nn/common.py (hunk 2 above)
Comment (baiyfbupt) [reply python/paddle/static/nn/common.py:240]: will fix in next PR

**Pull request merged**